modules/service/src/main/scala/lucuma/itc/service/ItcMappings.scala

```diff
@@ -379,34 +290,34 @@ object ItcMapping extends ItcCacheOrRemote with Version {
                 RootEffect.computeEncodable("spectroscopyIntegrationTime") { (p, env) =>
                   env
                     .getR[SpectroscopyIntegrationTimeInput]("input")
-                    .flatMap(toSpectroscopyTimeRequest)
-                    .flatTraverse(
+                    .flatMap(AsterismSpectroscopyTimeRequest.fromInput)
+                    .flatTraverse:
                       calculateSpectroscopyIntegrationTime(environment, redis, itc)
-                    )
                 },
                 RootEffect.computeEncodable("imagingIntegrationTime") { (p, env) =>
                   env
                     .getR[ImagingIntegrationTimeInput]("input")
-                    .flatMap(toImagingTimeRequest)
-                    .flatTraverse(
+                    .flatMap(AsterismImagingTimeRequest.fromInput)
+                    .flatTraverse:
                       calculateImagingIntegrationTime(environment, redis, itc)
-                    )
                 },
-                RootEffect.computeEncodable("optimizedSpectroscopyGraph") { (p, env) =>
+                RootEffect.computeEncodable("spectroscopyGraph") { (p, env) =>
                   env
-                    .getR[OptimizedSpectroscopyGraphInput]("input")
-                    .flatMap(toGraphRequest)
-                    .flatTraverse(
+                    .getR[SpectroscopyGraphInput]("input")
+                    .flatMap(AsterismGraphRequest.fromInput)
+                    .flatTraverse:
                       spectroscopyGraph(environment, redis, itc)
-                    )
                 },
                 RootEffect.computeEncodable("spectroscopyIntegrationTimeAndGraph") { (p, env) =>
                   env
                     .getR[SpectroscopyIntegrationTimeAndGraphInput]("input")
-                    .flatMap(u => toSpectroscopyTimeRequest(u).map((_, u.significantFigures)))
-                    .flatTraverse { case (tr, fig) =>
+                    .flatMap: input =>
+                      AsterismSpectroscopyTimeRequest
+                        .fromInput(input)
+                        .map((_, input.significantFigures))
+                    .flatTraverse: (tr, fig) =>
                       spectroscopyIntegrationTimeAndGraph(environment, redis, itc)(tr, fig)
-                    }
+                        .flatTap(x => Logger[F].info(x.toString))
```

Comment:
Logging?

Reply:
Thank you, there's still work to be done, including cleanup.